docs/cadl-for-openapi-dev.md

```diff
@@ -0,0 +1,285 @@
+| `type: string, format: date`      | `plainDate`      |                        |
+| `type: string, format: date-time` | `zonedDateTime`  | RFC 3339 date          |
+| `type: string, format: password`  | `@secret string` |                        |
+
+OpenAPI supports a variety of "assertions" that can be used further restrict the values allowed for a data type.
+These are actually borrowed into OpenAPI from JSON Schema.
+
+For `type: integer` and `type: number` data types:
+
+| OpenAPI/JSON Schema keyword | Cadl construct              | Notes |
+| --------------------------- | --------------------------- | ----- |
+| `minimum: value`            | `@minimum(value)` decorator |       |
+| `maximum: value`            | `@maximum(value)` decorator |       |
+
+For `type: string` data types:
+
+| OpenAPI/JSON Schema keyword | Cadl construct                | Notes |
+| --------------------------- | ----------------------------- | ----- |
+| `minLength: value`          | `@minLength(value)` decorator |       |
+| `maxLength: value`          | `@maxLength(value)` decorator |       |
+| `pattern: regex`            | `@format(regex)` decorator    |       |
+
+To define a `type: string` parameter or property with an `enum`, define the Cadl element as the union of the enum values.
+For example:
+
+```cadl
+status: "Running" | "Stopped" | "Failed"
+```
+
+## Info Object
+
+In OpenAPI, the `info` object [[v2][v2-info], [v3][v3-info]] contains metadata about the API such as a `title`, `description`, `license`, and `version`.
+
+[v2-info]: https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md#infoObject
+[v3-info]: https://github.com/OAI/OpenAPI-Specification/blob/3.0.3/versions/3.0.3.md#infoObject
+
+In Cadl this information is specified with [decorators on the namespace][cadl-service-metadata].
+
+| OpenAPI `info` field | Cadl decorator    | Notes                    |
+| -------------------- | ----------------- | ------------------------ |
+| `title`              | `@serviceTitle`   |                          |
+| `version`            | `@serviceVersion` |                          |
+| `description`        |                   | Not currently supported. |
+| `license`            |                   | Not currently supported. |
+| `contact`            |                   | Not currently supported. |
+
+[cadl-service-metadata]: https://github.com/microsoft/cadl/blob/main/docs/tutorial.md#service-definition-and-metadata
+
+## Consumes / Produces (OAS2)
+
+In OpenAPI v2, the top-level `consumes` and `produces` fields specify a list of MIME types an operation can consume / produce
+when not overridden by a `consumes` or `produces` on an individual operation.
+
+In Cadl, these global values for `consumes` and `produces` are defined with the `@consumes` and `@produces` decorators.
+
+```cadl
+@produces("application/json", "image/png")
+@consumes("application/json", "application/octet-stream")
```

Comment:
This needs to be applied on the global namespace right?

Reply:
Maybe, but the tutorial doesn't say anything about a "global" namespace. Does this need to be described in the tutorial, or is there some other word for it in the tutorial?  Or can we say "outer-most"? Could there be more than one "outer-most" namespace?  If so, what then?

Reply:
IIRC they go on the namespace bearing the service metadata, presumably an emitter could support multiple namespaces bearing this metadata via something like walking up from an operation to the nearest namespace containing the metadata (and emitting multiple  json files as necessary), but I suspect that doesn't work.